sagemaker/launch/launch_utils.py

```diff
@@ -0,0 +1,90 @@
+    https://sagemaker.readthedocs.io/en/stable/frameworks/pytorch/using_pytorch.html#model-directory-structure`_,
+    this function will put the entry point file into a sub-folder,named `code`, and then zip the
+    `model.bin`, the `**.yaml`, the `**.json` file, and the `code` sub-folder into a tar package
+    with the name specified by the wrap_name argument, and save it to the `output_path`.
+
+    This function will check if given files exist. Non-existance will raise an error.
+    
+    Parameters
+    ----------
+    path_to_model: str
+        The path of GraphStorm's `model.bin` file.
+    path_to_yaml: str
+        The path of the YAML file generated by GraphStorm during model training.
+    path_to_json: str
+        The path of JSON file generated by GraphStorm gconstruct or GSProcessing.
+    path_to_entry: str
+        The path of the entry point file for a specific task. The file will be put into a
+        sub-folder, named 'code'.
+    output_path: str
+        The folder where the output tar package will be saved. If not provided, will
+        raise an error.
+    output_tarfile_name: str
+        The name of the tar package. Default is `model`.
+
+    """
+    # check if files exist
+    assert os.path.exists(path_to_model), f'The model file, {path_to_model}, does not exist.'
+    assert os.path.isfile(path_to_model), f'The model file, {path_to_model}, should be a file \
+                                            path, but got a folder.'
+    assert os.path.exists(path_to_yaml), f'The model configuration YAML file, {path_to_yaml}, \
+                                           does not exist.'
+    assert os.path.isfile(path_to_yaml), f'The model configuration YAML file, {path_to_yaml}, \
+                                           should be a file path, but got a folder.'
+    assert os.path.exists(path_to_json), f'The graph metadata JSON file, {path_to_json}, does \
+                                           not exist.'
+    assert os.path.isfile(path_to_json), f'The graph metadata JSON file, {path_to_json}, should \
+                                           be a file path, but got a folder.'
+    assert os.path.exists(path_to_entry), f'The SageMaker entry point file, {path_to_entry}, does \
+                                           not exist.'
+    assert os.path.isfile(path_to_entry), f'The SageMaker entry point file, {path_to_entry}, \
+                                           should be a file path, but got a folder.'
+
+    # create the output folder if not exist.
+    assert os.path.isdir(output_path), 'Output path should be a folder name, but got a ' + \
+                                       f'file {output_path}.'
+    os.makedirs(output_path, exist_ok=True)
+    os.makedirs(os.path.join(output_path, 'code'), exist_ok=True)
+
+    # copy the artifacts files to output_path
+    shutil.copy(path_to_entry, os.path.join(output_path, 'code', os.path.basename(path_to_entry)))
```

Comment:
We are assuming the entry point is implemented in a single file.
How to handle the case when the entry_point imports modules from other local files?

Reply:
The entry point file will be executed in the endpoint docker image. As designed, all functions used in entry point file MUST be included in GraphStorm core code because entry point file are part of GraphStorm.

If later we need to include more files, we can modify this function with backward compatibility, i.e., check if given a string, a list of strings, or a folder, and proceed accordingly.
1. if a string, meaning just one file, copy it to the `code/` folder.
2. if a list of strings, meaning several files, copy them to the `code/` folder.
3. if a folder, meaning all files in that folder, copy all files in that folder to the `code/` folder.

So far, we can keep this simple version, and keep used functions in GraphStorm core code.